### PR TITLE
Add a minimal by_queue limit

### PR DESCRIPTION
**Pull request by svenfuchs**
Triggered by severe queue time issues on support for larger orgs that have lots of OSX jobs clogging up even boosted concurrency.

Only accessible through an env var at the moment, as we haven't had any product discussion about this.

### CODE REVIEW

**Comment by svenfuchs** [comment]
Works on org-staging:

```
I c0ecbe Received event job:finished job=547395 for svenfuchs
I c0ecbe Locking scheduler.owners-svenfuchs with: redis, ttl: 150s
I c0ecbe Evaluating jobs for owner group: user svenfuchs
I c0ecbe max jobs for user svenfuchs by default: 5
I c0ecbe max jobs for user svenfuchs by queue builds.macstadium6: 1
I c0ecbe user svenfuchs: total: 3, running: 0, queueable: 1
```

**Comment by svenfuchs** [comment]
This is now deployed on org/com production.

While adding this new strategy I found what I believe actually was a bug in the `by_repo` limit. This change represents the fix: https://github.com/travis-ci/travis-scheduler/pull/63/files#diff-16a83b8f1f8a9f1fb99b4e10ed93cda3L31

```
           def current
 -            state.running_by_repo(repo.id) + queued
 +            state.running_by_repo(repo.id) + selected.select { |j| j.repository_id == repo.id }.size
            end
```

... where `queued` previously was the number of currently selected jobs (`selected.size`).

If I understand this right then this means we previously have scheduled fewer jobs for repos that had a per-repo concurrency limit set in cases where there are several jobs waiting across multiple repositories, and one of them is limited.

In other words, in certain cases we might now schedule **more** jobs for repositories that have a per-repo limit set.
